crates/iota/src/fire_drill.rs

```diff
@@ -160,11 +161,16 @@ async fn update_next_epoch_metadata(
     new_config.protocol_key_pair =
         KeyPairWithPath::new(IotaKeyPair::Ed25519(new_protocol_key_pair));
 
-    let validators = iota_client
+    let iota_system_state = iota_client
         .governance_api()
         .get_latest_iota_system_state()
-        .await?
-        .active_validators;
+        .await?;
+    let validators = match iota_system_state {
+        IotaSystemStateSummary::V1(v1) => v1.active_validators,
+        IotaSystemStateSummary::V2(v2) => v2.active_validators,
+        _ => panic!("unsupported IotaSystemStateSummary"),
```

Comment:
All of these functions return a result. Should we do that here instead of panicking?

Reply:
We can definitely `bail`, but I do see some panics in `validator_commands` already. For example

https://github.com/iotaledger/iota/blob/466ac8285237eca2ab06a9dcf4721c61d5cf9af2/crates/iota/src/validator_commands.rs#L309-L312

So whatever you think is more consistent for the API. cc: @thibault-martinez

Reply:
Doesn't make a big difference for a CLI but maybe the error would be presented in a nicer way to the user by bailing, I'll do the change

Reply:
Actually, created an issue https://github.com/iotaledger/iota/issues/5506, let's merge this and adapt later